waspc/data/Generator/templates/server/src/dbClient.js

```diff
@@ -12,6 +13,41 @@ const createDbClient = () => {
   const prismaClient = new Prisma.PrismaClient()
 
   {=# isAuthEnabled =}
+
+  prismaClient.$use(async (params, next) => {
+    // Ensure strong plaintext password. Must come before hashing middleware.
+    // Throws an EntityValidationError on the first validation that fails.
+    if (params.model === '{= userEntityUpper =}' && params.action === 'create') {
+        const data = params.args.data || {}
+
+        const defaultValidations = [
+          { name: 'email must be present', fn: data => !!data.email },
+          { name: 'password must be present', fn: data => !!data.password },
+          { name: 'password must be at least 8 characters', fn: data => data.password.length >= 8 },
+          { name: 'password must contain a number', fn: data => /\d/.test(data.password) },
+        ]
+
+        let validations = params.args._waspSkipDefaultValidations ? [] : defaultValidations
+        if (Array.isArray(params.args._waspCustomValidations)) {
+          validations = validations.concat(params.args._waspCustomValidations)
+        }
```

Comment:
Just for fun, we could rewrite this to use `const` as:

```js
const validations = {
  ... (params.args._waspSkipDefaultValidations ? [] : defaultValidations),
  ... (Array.isArray(params.args._waspCustomValidations) ? params.args._waspCustomValidations : [])
}
```

or if you omit the check on the second one, it can be even shorter:

```js
const validations = {
  ... (params.args._waspSkipDefaultValidations ? [] : defaultValidations),
  ... (params.args._waspCustomValidations || [])
}
```

Is this more readable though? I think it is + I love immutability when achievable, but I am not strongly opinionated here.

Reply:
Yep, I think the last one looks good. My JS is rusty so thanks for the reminder of spread syntax. 😄 Minor typo I think on object wrapping it, but I got the gist! I will update to:
```
const validations = [
  ... (params.args._waspSkipDefaultValidations ? [] : defaultValidations),
  ... (params.args._waspCustomValidations || [])
]
```

Thanks!

Reply:
Ah yes nice catch! Mine is also a bit rusty hehe, but combined we can figure smth out :D.

Reply:
This has been addressed here: https://github.com/wasp-lang/wasp/pull/376/commits/3ffbc62c75cf4a43363b90fc8c2b1480309891ba Thanks!